src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Android.cs

```diff
@@ -0,0 +1,473 @@
+
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace System
+{
+    public sealed partial class TimeZoneInfo
+    {
+        private const string TimeZoneFileName = "tzdata";
+
+        // TODO: Consider restructuring underlying AndroidTimeZones class©.
+        // Although it may be easier to work with this way.
+        private static TimeZoneInfo GetLocalTimeZoneCore()
+        {
+            return AndroidTimeZones.Local!;
+        }
+
+        private static List<string> GetTimeZoneIds()
+        {
+            return new List<string>(AndroidTimeZones.GetAvailableIds());
+        }
+
+        //TODO: TryGetTimeZoneFromLocalMachine maps to FindSystemTimeZoneByIdCore in mono/mono implementation
+        private static TimeZoneInfoResult TryGetTimeZoneFromLocalMachineCore(string id, out TimeZoneInfo? value, out Exception? e)
+        {
+            value = null;
+            e = null;
+
+            try
+            {
+                // can id be null or empty here?
+                value = AndroidTimeZones.GetTimeZone(id, id);
+
+                // Are the below exceptions correct? Does GetTimeZone hit them at all
+            }
+            catch (UnauthorizedAccessException ex)
+            {
+                e = ex;
+                return TimeZoneInfoResult.SecurityException;
+            }
+            catch (FileNotFoundException ex)
+            {
+                e = ex;
+                return TimeZoneInfoResult.TimeZoneNotFoundException;
+            }
+            catch (DirectoryNotFoundException ex)
+            {
+                e = ex;
+                return TimeZoneInfoResult.TimeZoneNotFoundException;
+            }
+            catch (IOException ex)
+            {
+                e = new InvalidTimeZoneException(SR.Format(SR.InvalidTimeZone_InvalidFileData, id, GetTimeZoneDirectory() + TimeZoneFileName), ex);
+                return TimeZoneInfoResult.InvalidTimeZoneException;
+            }
+
+            if (value == null)
+            {
+                e = new InvalidTimeZoneException(SR.Format(SR.InvalidTimeZone_InvalidFileData, id, GetTimeZoneDirectory() + TimeZoneFileName));
+                return TimeZoneInfoResult.TimeZoneNotFoundException; // Mono/mono throws TimeZoneNotFoundException, runtime throws InvalidTimeZoneException
+            }
+
+            return TimeZoneInfoResult.Success;
+        }
+
+        private static class AndroidTimeZones
+        {
+            private static AndroidTzData? db = GetDefaultTimeZoneDB();
+
+            private static AndroidTzData? GetDefaultTimeZoneDB()
+            {
+                foreach (var p in AndroidTzData.Paths)
+                {
+                    if (File.Exists(p))
+                    {
+                        return new AndroidTzData(AndroidTzData.Paths);
+                    }
+                }
```

Comment:
This was mostly adapted from mono/mono, I am currently trying to refactor the code to be more direct, and any suggestions are welcome.

Reply:
I'd suggest removing `internal static readonly string[] Paths` because it's used only once and kept forever and initialize the data class with some default values even if none of the paths exist.

Reply:
Removed and opted to use logic to get tzdata file path and directory in `GetTimeZoneDirectory()`